src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1148,6 +1068,134 @@ public ValidateRangeAttribute(ValidateRangeKind kind) : base()
             _rangeKind = kind;
         }
 
+        private void ValidatePredefinedRange(object element)
+        {
+            Type commonType = GetCommonType((0).GetType(),element.GetType());
+            if (commonType == null)
+            {
+                    throw new ValidationMetadataException(
+                    "ValidationRangeElementType",
+                    null, 
+                    Metadata.ValidateRangeElementType,
+                    element.GetType().Name, 
+                    0.GetType().Name);
```

Comment:
`0.GetType()` and `0.GetType().Name` is good condidates for static read only constant.